src/java/org/apache/cassandra/locator/DynamicEndpointSnitch.java

```diff
@@ -154,31 +326,203 @@ private void registerMBean()
 
     public void close()
     {
-        updateSchedular.cancel(false);
-        resetSchedular.cancel(false);
+        if (updateScoresScheduler != null)
+            updateScoresScheduler.cancel(false);
+        if (updateSamplesScheduler != null)
+            updateSamplesScheduler.cancel(false);
+
+        for (AnnotatedMeasurement measurement : samples.values())
+        {
+            if (measurement.probeFuture != null)
+                measurement.probeFuture.cancel(false);
+
+            measurement.millisSinceLastMeasure.set(0);
+            measurement.millisSinceLastRequest.set(MAX_PROBE_INTERVAL_MS);
+            measurement.probeTimerMillis = 0;
+        }
 
         MBeanServer mbs = ManagementFactory.getPlatformMBeanServer();
         try
         {
-            mbs.unregisterMBean(new ObjectName(mbeanName));
+            if (mbeanRegistered)
+                mbs.unregisterMBean(new ObjectName(mbeanName));
         }
         catch (Exception e)
         {
             throw new RuntimeException(e);
         }
     }
 
+    /**
+     * Background task running on the samples dictionary. The default implementation sends latency probes (PING)
+     * messages to explore nodes that we have not received timings for recently but have ranked in
+     * {@link DynamicEndpointSnitch#sortedByProximity(InetAddressAndPort, ReplicaCollection)}.
+     */
+    protected void updateSamples()
+    {
+        // Split calculation of probe timers from sending probes for testability
+        calculateProbes(samples, dynamicLatencyProbeInterval);
+
+        if (!StorageService.instance.isGossipActive())
```

Comment:
This needs to be here so that the bookeeping clocks keep ticking. Basically we do the calculation but don't send any probes (respecting a disabled gossip indicating we shouldn't be sending messages).